api/VERSION

```diff
@@ -1 +1 @@
-v2
+v3
```

Comment:
api_version in google3 is `v1` and here is `v2`, so no need to increment before next import?

Reply:
I worry this will be confusing to have to check the version every time. It could result in difficulties if we ever revert a CL that changes this number. 

As discussed with Wayne, let's always try to increment the number in each CL. This should be OK for API Gateway.

Reply:
It makes sense from our side. Just one thing, in google3 it would be like v1 -> v4 -> v7 -> v14 and can we confirm with API Gateway? @toddbeckman-g

Reply:
For each breaking change, we should increment this version so that github history of this file will show all the breaking changes.

Reply:
It doesn't look like this PR has any breaking API changes which affect API Gateway, so I think no increment is necessary.

To answer your question though, it doesn't really matter what the version number here is, as long as we can differentiate between them, so it is okay to skip version numbers.

Reply:
Thanks!

Reply:
It's a breaking API change. Older filter configs generate no-op backend routing rules. With this change, envoy will not accept configs with no-op rules.

Reply:
Let me clarify. Backend routing filter only re-writes the path portion of a URI. We use envoy route config to re-write the host portion.

So even for API Gateway, some operations will not need backend routing filter. If the only change we need to make is the hostname but not the path, requests can skip over this filter.

That is the breaking API change. Beforehand, those requests with no path re-write would go into the backend routing filter and be translated as APPEND requests. Now, they skip over the logic in the filter.

Reply:
Ah I see now. Then yes, I agree that this is a breaking change. Sorry for the confusion.